File: app/src/main/java/ao/co/r4c/model/Solicitacao.java
package ao.co.r4c.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Solicitacao {

    @Expose
    @SerializedName("id")
    private int id;

    @Expose
    @SerializedName("id_motorista")
    private int id_motorista;


    @Expose
    @SerializedName("nome_motorista")
    private String nome_motorista;

    @Expose
    @SerializedName("id_passageiro")
    private int id_passageiro;

    @Expose
    @SerializedName("nome_passageiro")
    private String nome_passageiro;


    @Expose
    @SerializedName("minutos")
    private String minutos;

    @Expose
    @SerializedName("endereco")
    private String endereco;


    @Expose
    @SerializedName("distancia")
    private float distancia;

    @Expose
    @SerializedName("lat_origem")
    private float lat_origem;

    @Expose
    @SerializedName("lon_origem")
    private float lon_origem;

    @Expose
    @SerializedName("lat_destino")
    private float lat_destino;

    @Expose
    @SerializedName("lon_destino")
    private float lon_destino;

    @Expose
    @SerializedName("descricao")
    private String descricao;

    @Expose
    @SerializedName("estado_solicitacao")
    private Boolean estado_solicitacao;


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId_motorista() {
        return id_motorista;
    }

    public void setId_motorista(int id_motorista) {
        this.id_motorista = id_motorista;
    }

    public String getNome_motorista() {
        return nome_motorista;
    }

    public void setNome_motorista(String nome_motorista) {
        this.nome_motorista = nome_motorista;
    }

    public int getId_passageiro() {
        return id_passageiro;
    }

    public void setId_passageiro(int id_passageiro) {
        this.id_passageiro = id_passageiro;
    }

    public String getNome_passageiro() {
        return nome_passageiro;
    }

    public void setNome_passageiro(String nome_passageiro) {
        this.nome_passageiro = nome_passageiro;
    }

    public String getMinutos() {
        return minutos;
    }

    public void setMinutos(String minutos) {
        this.minutos = minutos;
    }

    public String getEndereco() {
        return endereco;
    }

    public void setEndereco(String endereco) {
        this.endereco = endereco;
    }

    public float getDistancia() {
        return distancia;
    }

    public void setDistancia(float distancia) {
        this.distancia = distancia;
    }

    public float getLat_origem() {
        return lat_origem;
    }

    public void setLat_origem(float lat_origem) {
        this.lat_origem = lat_origem;
    }

    public float getLon_origem() {
        return lon_origem;
    }

    public void setLon_origem(float lon_origem) {
        this.lon_origem = lon_origem;
    }

    public float getLat_destino() {
        return lat_destino;
    }

    public void setLat_destino(float lat_destino) {
        this.lat_destino = lat_destino;
    }

    public float getLon_destino() {
        return lon_destino;
    }

    public void setLon_destino(float lon_destino) {
        this.lon_destino = lon_destino;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public Boolean getEstado_solicitacao() {
        return estado_solicitacao;
    }

    public void setEstado_solicitacao(Boolean estado_solicitacao) {
        this.estado_solicitacao = estado_solicitacao;
    }
}

File: backend/api/objects/solicitacao.php
<?php

class Solicitacao
{
    //Conexão do banco de dados e nome da tabela
    private $connection;
    private $table_name = "tb_solicitacoes";

    //Propriedades do objecto
    private $id;
    private $id_motorista;
    private $nome_motorista;
    private $id_passageiro;
    private $nome_passageiro;
    private $minutos;
    private $id_local_origem;
    private $id_local_destino;

    //Construtor que recebe como parâmetro a conexão
    function __construct($connection)
    {
        $this->connection = $connection;
    }

    /*Permite inserir nova solicitação*/
    function inserir_solicitacao()
    {
        //Insert query
        $query = "INSERT INTO " . $this->table_name . " (id_motorista, id_passageiro, id_local_origem, id_local_destino, tempo_solicitacao)
        values ( ?, ?, ?, ?, current_time)";


        //Prepare query
        if ($stmt = $this->connection->prepare($query)) {

            //Sanitize
            $this->id_passageiro = htmlspecialchars(strip_tags($this->id_passageiro));
            $this->id_motorista = htmlspecialchars(strip_tags($this->id_motorista));
            $this->id_local_origem = htmlspecialchars(strip_tags($this->id_local_origem));
            $this->id_local_destino = htmlspecialchars(strip_tags($this->id_local_destino));

            //Bind parameters
            $stmt->bind_param("iiii", $this->id_motorista, $this->id_passageiro, $this->id_local_origem, $this->id_local_destino);

            //Execute query
            if ($stmt->execute())
                return true;
        }

        return false;
    }


    //Permite listar todos os usuarios activos
    function verificarSolicitacao()
    {
        //Query to list record
        $query = " select id,
                          id_motorista, (select nome from tb_usuarios where id = tb_solicitacoes.id_motorista) as motorista,
                          id_passageiro, (select nome from tb_usuarios where id=tb_solicitacoes.id_passageiro) as passageiro,
                          minute(CURRENT_TIME ())-minute(tempo_solicitacao) as minutos, lat_origem, lon_origem, lat_destino, lon_destino,
                          estado_solicitacao, descricao
                          from " . $this->table_name . " where id_motorista=?

                          ";

        //Prepare query
        $stmt = $this->connection->prepare($query);

        //Sanitize
        $this->id_motorista = htmlspecialchars(strip_tags($this->id_motorista));

        $stmt->bind_param("i", $this->id_motorista);

        //Execute query
        $stmt->execute();

        return $stmt;
    }

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getIdLocalDestino()
    {
        return $this->id_local_destino;
    }

    /**
     * @param mixed $id_local_destino
     */
    public function setIdLocalDestino($id_local_destino)
    {
        $this->id_local_destino = $id_local_destino;
    }

    /**
     * @return mixed
     */
    public function getIdLocalOrigem()
    {
        return $this->id_local_origem;
    }

    /**
     * @param mixed $id_local_origem
     */
    public function setIdLocalOrigem($id_local_origem)
    {
        $this->id_local_origem = $id_local_origem;
    }

    /**
     * @return mixed
     */
    public function getIdMotorista()
    {
        return $this->id_motorista;
    }

    /**
     * @param mixed $id_motorista
     */
    public function setIdMotorista($id_motorista)
    {
        $this->id_motorista = $id_motorista;
    }

    /**
     * @return mixed
     */
    public function getIdPassageiro()
    {
        return $this->id_passageiro;
    }

    /**
     * @param mixed $id_passageiro
     */
    public function setIdPassageiro($id_passageiro)
    {
        $this->id_passageiro = $id_passageiro;
    }

    /**
     * @return mixed
     */
    public function getMinutos()
    {
        return $this->minutos;
    }

    /**
     * @param mixed $minutos
     */
    public function setMinutos($minutos)
    {
        $this->minutos = $minutos;
    }

    /**
     * @return mixed
     */
    public function getNomeMotorista()
    {
        return $this->nome_motorista;
    }

    /**
     * @param mixed $nome_motorista
     */
    public function setNomeMotorista($nome_motorista)
    {
        $this->nome_motorista = $nome_motorista;
    }

    /**
     * @return mixed
     */
    public function getNomePassageiro()
    {
        return $this->nome_passageiro;
    }

    /**
     * @param mixed $nome_passageiro
     */
    public function setNomePassageiro($nome_passageiro)
    {
        $this->nome_passageiro = $nome_passageiro;
    }

    /**
     * @return string
     */
    public function getTableName()
    {
        return $this->table_name;
    }

    /**
     * @param string $table_name
     */
    public function setTableName($table_name)
    {
        $this->table_name = $table_name;
    }



}